### PR TITLE
Update nemo-rl to latest

### PR DESCRIPTION
**Pull request by smahdavi4**
Upgrading nemo-rl to the latest main:
* Vllm version 11.0
* Full support for async RL
* RL support with GB200 clusters.
* Config changes to align with nemo-rl and fix some of the bugs on our side (such as clipping)

Limitations:
The gradient clipping is not unified yet, we can merge after [this](https://github.com/NVIDIA-NeMo/RL/pull/1618) is resolved.

## Summary by CodeRabbit

* **New Features**
  - Asynchronous GRPO training mode with configurable weight update controls
  - LoRA and PEFT support for parameter-efficient fine-tuning

* **Improvements**
  - Updated build infrastructure with enhanced CUDA architecture targeting
  - Extended GRPO and SFT training configurations with expanded hyperparameter options
  - Enhanced generation and optimization parameter controls for improved performance

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: nemo_skills/training/nemo_rl/configs/grpo.yaml [hunk 1]
```diff
@@ -191,10 +155,10 @@ policy:
       start_weight_decay: ${policy.megatron_cfg.optimizer.weight_decay}
       end_weight_decay: ${policy.megatron_cfg.optimizer.weight_decay}
       weight_decay_incr_style: "constant"
-      lr_decay_style: "cosine"
-      lr_decay_iters: ${grpo.max_num_steps}
-      lr_warmup_iters: 0
-      lr_warmup_init: 1.0e-6
+      lr_decay_style: "constant"
```
Comment (Kipok) [inline nemo_skills/training/nemo_rl/configs/grpo.yaml]: let's maybe keep original defaults? Otherwise it will be harder to set LR decay. While cosine can work as both constant schedule as well as decay by just adjusting min lr

File: nemo_skills/training/nemo_rl/configs/grpo.yaml [hunk 2]
```diff
@@ -204,7 +168,53 @@ policy:
...
+      lr: ${policy.lr}
+      weight_decay: ${policy.weight_decay}
+      betas: [0.9, 0.999]
+      eps: 1e-8
+      # when using Dtensor, we need to set foreach
+      # and fused to False
+      foreach: False
+      fused: False
+
+  scheduler:
```
Comment (Kipok) [inline nemo_skills/training/nemo_rl/configs/grpo.yaml:207]: and same here, ideally keep original

File: nemo_skills/training/nemo_rl/configs/sft-legacy-85eeb8d.yaml [hunk 3]
```diff
@@ -0,0 +1,197 @@
+# SFT Algorithm Configuration
```
Comment (Kipok) [inline nemo_skills/training/nemo_rl/configs/sft-legacy-85eeb8d.yaml]: Should we remove these? The configs can always be found in git history, not really sure if there is much point in explicitly keeping them as separate files

**Review by Kipok — APPROVED** [review]
